core-plugins/src/main/java/co/cask/hydrator/plugin/common/AvroToStructuredTransformer.java

```diff
@@ -71,4 +74,26 @@ public StructuredRecord transform(StructuredRecord.Builder builder, GenericRecor
 
     return builder.build();
   }
+
+  public StructuredRecord transform(GenericRecord genericRecord, String addFieldKey, String addFieldValue)
```

Comment:
This isn't a very natural place for this logic, as this class should be about converting records. Modify the other transform method to return a StructuredRecord.Builder and remove the builder as an argument. The logic to add a field to the schema should be in the input format class, not here.